clients/ui/api/openapi/mod-arch.yaml

```diff
@@ -1972,6 +2142,84 @@ components:
           additionalProperties:
             $ref: "#/components/schemas/FilterOption"
 
+    # Schema for catalog settings
+    CatalogSourceConfig:
+      description: A single catalog source configuration
+      type: object
+      allOf:
+        - $ref: "#/components/schemas/CatalogSourceConfigPayload"
+    CatalogSourceConfigList:
+      description: List of catalog sources.
+      type: object
+      properties:
+        catalogs:
+          description: Array of catalog sources.
+          type: array
+          items:
+            $ref: "#/components/schemas/CatalogSourceConfig"
+
+    # scehma for model catalog source payload
+    CatalogSourceConfigPayload:
+      description: Payload for creating the catalog source
+      type: object
+      required:
+        - name
+        - id
+        - type
+      properties:
+        name:
+          type: string
+          description: The name of the catalog source.
+          example: Sample Catalog
+        id:
+          type: string
+          description: A unique identifier for a `CatalogSource`.
+          example: sample_custom_catalog
+        type:
+          type: string
+          example: yaml
+        enabled:
+          type: boolean
+          description: Whether the catalog source is enabled.
+          default: true
+          example: true
+        labels:
+          description: |-
+            Labels associated with the source. The catalog source labels can be
+            used to filter models from sources with the same label (see
+            `sourceLabel` on `/api/model_catalog/v1alpha1/models`)
+          type: array
+          items:
+            type: string
+        isDefault:
+          type: boolean
+          example: true
+        properties:
+          type: object
+          properties:
+            yamlCatalogPath:
+              type: string
+              example: sample-catalog.yaml
+            apiKey:
+              type: string
+              description: HuggingFace access token - should be stored in a Kubernetes secret and referenced here, or set via environment variable
+              example: "apiKey"
```

Comment:
The description here isn't quite right for this layer. This apiKey is another example of how we'll need the BFF to provide abstraction - the UI should actually directly pass the value here, and then later when the BFF updates the configmap it will need to deal with creating/updating a k8s secret and referencing the secret in the configmap. Then when the BFF is fetching a CatalogSourceConfig it will need to go fetch that secret and return its value directly in the CatalogSourceConfig.